### PR TITLE
Incorporate Custom Role API changes - include created and lastupdated…

### PR DESCRIPTION
**Pull request by nateyocom-okta**
## Description:
- Incorporate changes to beta custom role api - include created and lastUpdated properties on Custom Roles, Resource Sets and Members, and make consistent the use of links on roles to point to permissions and vice versa

- Goes with monolith release CF 4-9-2021 (so for the 2021.04.1 release)

### Resolves:

* [OKTA-385063](https://oktainc.atlassian.net/browse/OKTA-385063)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md [hunk 1]
```diff
@@ -2872,22 +3069,35 @@ A Custom Role is a custom set of [Permissions](#permission-types). A Custom Role
...
 ### Permission types
 
 <ApiLifecycle access="beta" />
 
-Permissions can be used to build Custom Roles. Permissions to manage a resource also grant the viewing privileges for the same resource so that you won't need to assign them separately.
+Permissions can be added to, or removed from, custom Roles.  Permissions to manage a resource also grant the viewing privileges for the same resource so that you won't need to assign them separately.
 
 User permissions are only effective with respect to the Group(s) to which the admin is granted a Role through Resource Set assignments.
 
+### Permission properties
```
Comment (alifatolahi-okta) [inline packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md:3092]: Are these permission properties or role properties? I think it's a bit confusing to attribute timestamp to permission. We should at least clarify whether `created` and `lastUpdated` refer to the timestamp of the role-permision association or the role

**Review by alifatolahi-okta — COMMENTED** [review]
Just one comment about attributing timestamps to permission objects

File: packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md (hunk 1 above)
Comment (nateyocom-okta) [reply packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md:3092]: I'll also make this a #### - i.e. a subsection of Permission types for clarity

**Review by alifatolahi-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md [hunk 2]
```diff
@@ -130,12 +134,70 @@ curl -v -X GET \
...
+}
+```
+
+### Update Role
+
+<ApiLifecycle access="beta" />
+
+<ApiOperation method="put" url="/api/v1/iam/roles/${roleIdOrLabel}" />
+
+Update a custom Role's label or description by its ID or label
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md:156]:
```suggestion
Updates a custom Role's label or description by its ID or label
```

File: packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md [hunk 3]
```diff
@@ -268,6 +393,55 @@ curl -v -X POST \
...
+#### Request parameters
+
+| Parameter      | Description                          | Param Type   | DataType                              | Required |
+| :------------- | :----------------------------------- | :----------- | :------------------------------------ | :------- |
+| `roleIdOrLabel`  | `id` or `label` of the Role          | URL          | String                                | TRUE     |
+| `permissionType` | Permission to retrieve from the Role        | URL          | [Permission type](#permission-types) | TRUE     |
+
+#### Response parameters
+
+The requested permission object.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md:414]:
```suggestion
The requested permission object
```

File: packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md [hunk 4]
```diff
@@ -2960,33 +3191,25 @@ The ID of a resource is unique to the Resource Set, whereas the link that points
...
+        },
+        "groups": {
           "href": "https://${yourOktaDomain}/api/v1/groups"
         }
       }
     }
 ```
 
-> **Note:** When you provide either [all Groups](#all-groups-as-resource) and [all Users](#all-users-as-resource), it results in including all Users and Groups.
+> **Note:** When you provide either [all Groups](#all-groups-as-resource) or [all Users](#all-users-as-resource) when adding a resource to a [Resource Set](#resource-set-object), it results in including a single resource which encompasses all Users and Groups.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/iam-roles/index.md:3212]:
```suggestion
> **Note:** When you provide either [all Groups](#all-groups-as-resource) or [all Users](#all-users-as-resource) when adding a resource to a [Resource Set](#resource-set-object), a single resource that encompasses all Users and Groups is included.
```

**Review by susanharper-okta — COMMENTED** [review]
two small nits and then one edit to the note.

**Review by susanharper-okta — APPROVED** [review]
LGTM

**Pull request merged**